massa-network/src/node_worker.rs

```diff
@@ -179,17 +179,47 @@ impl NodeWorker {
         })?;
         let write_timeout = self.cfg.message_timeout;
         let node_id_copy = self.node_id;
+        let storage = self.storage.clone();
         let node_writer_handle = tokio::spawn(async move {
             loop {
                 match writer_command_rx.recv().await {
                     Some(to_send) => {
                         let bytes_vec: Vec<u8> = match to_send {
                             ToSend::Msg(msg) => msg.to_bytes_compact().unwrap(),
                             ToSend::Block(block_id) => {
-                                break;
+                                // TODO: precisely allocate?
+                                let mut res: Vec<u8> = Vec::new();
+                                res.extend(u32::from(MessageTypeId::Block).to_varint_bytes());
+
+                                let block = storage.retrieve_block(&block_id).unwrap();
+                                let mut stored_block = block.write();
+                                if let Some(serialized) = stored_block.serialized.as_ref() {
+                                    res.extend(serialized);
+                                } else {
+                                    let serialized = stored_block.block.to_bytes_compact()?;
+                                    res.extend(&serialized);
+                                    stored_block.serialized = Some(serialized);
+                                }
+
+                                res
                             }
                             ToSend::Header(block_id) => {
-                                break;
+                                // TODO: precisely allocate?
```

Comment:
We don't know the size of the header, but it's computed each time we create a block, maybe event if we don't know its size it may be worth to have at least some heuristics.